test/lib/flat_mutation_reader_assertions.hh

```diff
@@ -543,8 +544,30 @@ class flat_reader_assertions_v2 {
     flat_mutation_reader_v2 _reader;
     dht::partition_range _pr;
     bool _ignore_deletion_time = false;
+    tombstone _rt{};
 private:
+    mutation_fragment_v2* peek_next() {
+        while (auto next = _reader.peek().get0()) {
+            // There is no difference between an empty row and a row that doesn't exist.
+            // While readers that emit spurious empty rows may be wasteful, it is not
+            // incorrect to do so, so let's ignore them.
+            if (next->is_clustering_row() && next->as_clustering_row().empty()) {
+                testlog.trace("Received empty clustered row: {}", mutation_fragment_v2::printer(*_reader.schema(), *next));
+                _reader().get();
+                continue;
+            }
+            // silently ignore spurious closing rtcs
+            if (!_rt && next->is_range_tombstone_change() && !next->as_range_tombstone_change().tombstone()) {
```

Comment:
A more generic condition:
```
next->is_range_tombstone_change() && next->as_range_tombstone_change().tombstone() == _rt
```
This also covers range tombstone changes, not actually representing a change (repeating the current change).